packages/engine/src/datastore/table/state/mod.rs

```diff
@@ -149,14 +161,20 @@ impl ExState {
         &mut self.inner.local_meta
     }
 
-    pub fn finalize_agent_pool(
+    // TODO can this be moved into ContextMut
+    /// Copies the current agent state into the Context before running state packages, which
+    /// stores a snapshot of state at the end of the last step
+    ///
+    /// This can result in a change in the number of groups and batches within the Context,
+    /// and thus it updates the group start indices registered in self
```

Comment:
```suggestion
    /// Copies the current agent state into the `Context` before running state packages, which
    /// stores a snapshot of state at the end of the last step.
    ///
    /// This can result in a change in the number of groups and batches within the `Context`,
    /// and thus it updates the group start indices registered in self.
```